pd/terminate_monitor_hook.cpp

```diff
@@ -6,8 +6,19 @@ bool terminate_monitor_hook::add_redirect(unsigned __int64 target_address)
 {
 	SIZE_T num_written = 0;
 	unsigned char inject[0x20];
+	int num_write = 0;
 
-	if (!_is64)

```

Comment:
Awesome job here, thanks for this change! I have no recollection as to why I wrote such a long hook patch here, hah.

For this line, I think on a 64bit OS when hooking a 32bit process it will still need to apply the 32bit patch. As I understand your pull request, _options->Win64 is the OS, not the process mode.

I think this line should still be:
if (!_is64)

Which is the mode of the process module we are hooking. Did you find this change was needed for it to work for some reason?

Reply:
> For this line, I think on a 64bit OS when hooking a 32bit process it will still need to apply the 32bit patch. As I understand your pull request, _options->Win64 is the OS, not the process mode.
>
```c
if (!_options->Win64) {
    // 32bit OS, 32bit Process
    // There is a problem that ZwTerminateThread will be broken
}
else if (!_is64) {
    // 64bit OS, 32bit Process
    // No problem here
}
else {
    // 64bit OS, 64bit Process
    // No problem here
}
```
It still can work when hooking a 32bit process on 64bit OS if we keep origin code, but I think it's ok to apply the near jmp patch to hook 32bit/64bit process on 64bit OS. The reason that I only apply the patch on 32bit process on 32bit OS is I don't know whether there is a reason to do the far jmp for process on 64bit OS. If you want me to patch them, I can try to do that.

Reply:
Thanks, I get it now! By any chance would you be up for updating the 32bit and 64bit process hooks to all follow your new hook format? Not just 32bit under 32bit OS. It might solve some other issues with the crashing of processes - eg csrss.exe process crashing.

```
if (!_is64) {
    **Your new shorter 32-bit jmp**
}
else {
    **Your new shorter 64-bit jmp**
}
```

Reply:
Check the update (0cf3ed5). It works fine both on 32/64bit OS

The csrss.exe still crashes, I think this problem still needs to do some additional research to figure it out why it crashes.